src/standalone/Soil/energy_hydrology.jl

```diff
@@ -230,6 +221,136 @@ function ClimaLand.make_compute_exp_tendency(
     return compute_exp_tendency!
 end
 
+"""
+    make_compute_imp_tendency(model::EnergyHydrology)
+
+An extension of the function `make_compute_imp_tendency`, for the integrated
+soil energy and heat equations, including phase change.
+
+This version of this function computes the right hand side of the PDE for
+`Y.soil.ϑ_l`, which is the only quantity we currently step implicitly.
+
+This has been written so as to work with Differential Equations.jl.
+"""
+function ClimaLand.make_compute_imp_tendency(
+    model::EnergyHydrology{FT},
+) where {FT}
+    function compute_imp_tendency!(dY, Y, p, t)
+        z = ClimaCore.Fields.coordinate_field(model.domain.space.subsurface).z
+        rre_top_flux_bc = p.soil.top_bc.water
+        rre_bottom_flux_bc = p.soil.bottom_bc.water
+
+        interpc2f = Operators.InterpolateC2F()
+        gradc2f = Operators.GradientC2F()
+
+        # Without topography only
+        # In Cartesian coordinates, W (z^) = Cov3 (z^)= Contra3 (n^ = z^)
+        # In spherical coordinates, W (r^) = Cov3 (r^) = Contra3 (n^ = r^)
+        # Passing WVector to gradient BC is passing a normal flux.
+
+        # Richards-Richardson RHS
+        divf2c_rre = Operators.DivergenceF2C(
+            top = Operators.SetValue(Geometry.WVector.(rre_top_flux_bc)),
+            bottom = Operators.SetValue(Geometry.WVector.(rre_bottom_flux_bc)),
+        )
+        # GradC2F returns a Covariant3Vector, so no need to convert.
+        @. dY.soil.ϑ_l =
+            -(divf2c_rre(-interpc2f(p.soil.K) * gradc2f(p.soil.ψ + z)))
+
+        # Don't update the prognostic variables we're stepping explicitly
+        @. dY.soil.θ_i = 0
+        @. dY.soil.ρe_int = 0
+    end
+    return compute_imp_tendency!
+end
+
+"""
+    ClimaLand.make_update_jacobian(model::EnergyHydrology{FT}) where {FT}
+
+Creates and returns the update_jacobian! function for the EnergyHydrology model.
+This updates the contribution for the soil liquid water content only.
+
+Using this Jacobian with a backwards Euler timestepper is equivalent
+to using the modified Picard scheme of Celia et al. (1990).
+"""
+function ClimaLand.make_update_jacobian(model::EnergyHydrology{FT}) where {FT}
+    function update_jacobian!(jacobian::ImplicitEquationJacobian, Y, p, dtγ, t)
+        (; matrix) = jacobian
+        (; ν, hydrology_cm, S_s, θ_r) = model.parameters
+
+        # Create divergence operator
```

Comment:
Maybe we can open an issue to note partial code duplication between this and rre make_update_jacobian?

Reply:
good point! I opened an issue here https://github.com/CliMA/ClimaLand.jl/issues/659